SalesforcePluginApp.ts

```diff
@@ -0,0 +1,831 @@
+                            };
+
+                            http
+                              .post(
+                                `${rocketChatServerUrl}api/v1/users.setStatus`,
+                                setStatusHttpRequest,
+                              )
+                              .then(async (statusResponse) => {
+                                console.log(
+                                  'Setting Salesforce Bot Status, Response:',
+                                );
+                                console.log(statusResponse);
+
+                                const assoc = new RocketChatAssociationRecord(
+                                  RocketChatAssociationModel.ROOM,
+                                  message.room.id,
+                                );
+                                await persistence.createWithAssociation(
+                                  sessionIdParsedResponse,
+                                  assoc,
+                                );
+
+                                const roomId = message.room.id;
+                                const room: ILivechatRoom = (await read
+                                  .getRoomReader()
+                                  .getById(roomId)) as ILivechatRoom;
+                                // tslint:disable-next-line: max-line-length
+                                const targetDepartment: IDepartment = (await read
+                                  .getLivechatReader()
+                                  .getLivechatDepartmentByIdOrName(
+                                    targetDeptName,
+                                  )) as IDepartment;
+
+                                const transferData: ILivechatTransferData = {
+                                  currentRoom: room,
+                                  targetDepartment: targetDepartment.id,
+                                  targetAgent: targetagent,
+                                };
+
+                                await modify
+                                  .getUpdater()
+                                  .getLivechatUpdater()
+                                  .transferVisitor(LcVisitor, transferData);
+                              })
+                              .catch((statusErr) => {
+                                console.log(
+                                  'Setting Salesforce Bot Status, Error:',
+                                );
+                                console.log(statusErr);
+                              });
+                          })
+                          .catch((loginErr) => {
+                            console.log(
+                              'Performing Salesforce Bot Login, Error:',
+                            );
+                            console.log(loginErr);
+                          });
+                      }
+                    } catch (err) {
+                      console.log('Perfoming Handoff, Error:');
```

Comment:
Not sure whats going on on my side... it says it successfully sets the status as online, but then fails to handover because the agent is 'offline'

```
I20200625-13:03:30.241(-4)? The App 0d3ac5b3-dd0b-43d3-924a-5a157b748d10 is requesting from the outter webs: {
I20200625-13:03:30.241(-4)?   appId: '0d3ac5b3-dd0b-43d3-924a-5a157b748d10',
I20200625-13:03:30.241(-4)?   method: 'post',
I20200625-13:03:30.242(-4)?   url: 'http://localhost:3000/api/v1/users.setStatus',
I20200625-13:03:30.242(-4)?   request: {
I20200625-13:03:30.242(-4)?     headers: {
I20200625-13:03:30.242(-4)?       'X-Auth-Token': 'N0cuz9LSKDb00x7joq9LqqdoB6EV3AfH4JERpmR7p5S',
I20200625-13:03:30.242(-4)?       'X-User-Id': 'GDbwZTSnF2hXJaLeu'
I20200625-13:03:30.242(-4)?     },
I20200625-13:03:30.242(-4)?     data: { message: 'online', status: 'online' },
I20200625-13:03:30.242(-4)?     params: {},
I20200625-13:03:30.242(-4)?     content: '{"message":"online","status":"online"}',
I20200625-13:03:30.242(-4)?     npmRequestOptions: {}
I20200625-13:03:30.242(-4)?   }
I20200625-13:03:30.243(-4)? }
I20200625-13:03:30.256(-4)? Setting Salesforce Bot Status, Response:
I20200625-13:03:30.257(-4)? {
I20200625-13:03:30.257(-4)?   statusCode: 200,
I20200625-13:03:30.257(-4)?   content: '{\n  "success": true\n}',
I20200625-13:03:30.257(-4)?   headers: {
I20200625-13:03:30.257(-4)?     'x-xss-protection': '1',
I20200625-13:03:30.257(-4)?     'x-instance-id': 'wrkqZy9JqnubebaXg',
I20200625-13:03:30.257(-4)?     'cache-control': 'no-store',
I20200625-13:03:30.257(-4)?     pragma: 'no-cache',
I20200625-13:03:30.257(-4)?     'content-type': 'application/json',
I20200625-13:03:30.257(-4)?     vary: 'Accept-Encoding',
I20200625-13:03:30.258(-4)?     date: 'Thu, 25 Jun 2020 17:03:30 GMT',
I20200625-13:03:30.258(-4)?     connection: 'close',
I20200625-13:03:30.258(-4)?     'transfer-encoding': 'chunked'
I20200625-13:03:30.258(-4)?   },
I20200625-13:03:30.258(-4)?   data: { success: true }
I20200625-13:03:30.290(-4)? }
I20200625-13:03:30.290(-4)? The App 0d3ac5b3-dd0b-43d3-924a-5a157b748d10 is storing a new object in their persistence that is associated with some models. {
I20200625-13:03:30.290(-4)?   key: '4cf9f96f-671a-47f0-ba2c-156101bb9ab6!1593104607351!HykWZI9foZQwWkVwMgq1zKV7NLQ=',
I20200625-13:03:30.291(-4)?   id: '4cf9f96f-671a-47f0-ba2c-156101bb9ab6',
I20200625-13:03:30.291(-4)?   clientPollTimeout: 40,
I20200625-13:03:30.291(-4)?   affinityToken: '13051fa0'
I20200625-13:03:30.291(-4)? } [
I20200625-13:03:30.291(-4)?   RocketChatAssociationRecord {
I20200625-13:03:30.291(-4)?     model: 'room',
I20200625-13:03:30.291(-4)?     id: 'cYEXRi2r8grHCjtmx'
I20200625-13:03:30.291(-4)?   }
I20200625-13:03:30.291(-4)? ]
I20200625-13:03:30.291(-4)? The App 0d3ac5b3-dd0b-43d3-924a-5a157b748d10 is getting the roomById: "cYEXRi2r8grHCjtmx"
I20200625-13:03:30.291(-4)? The App 0d3ac5b3-dd0b-43d3-924a-5a157b748d10 is looking for livechat departments.
I20200625-13:03:30.292(-4)? The App 0d3ac5b3-dd0b-43d3-924a-5a157b748d10 is transfering a livechat.
I20200625-13:03:30.292(-4)? Setting Salesforce Bot Status, Error:
I20200625-13:03:30.301(-4)? errorClass [Error]: User is offline [error-user-is-offline]
I20200625-13:03:30.301(-4)?     at app/livechat/server/lib/Helper.js:198:9
I20200625-13:03:30.301(-4)?     at /Users/dlassalle/.meteor/packages/promise/.0.11.2.1e1wn8z.joczg++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40 {
I20200625-13:03:30.301(-4)?   isClientSafe: true,
I20200625-13:03:30.301(-4)?   error: 'error-user-is-offline',
I20200625-13:03:30.301(-4)?   reason: 'User is offline',
I20200625-13:03:30.301(-4)?   details: { function: 'forwardRoomToAgent' },
I20200625-13:03:30.301(-4)?   message: 'User is offline [error-user-is-offline]',
I20200625-13:03:30.302(-4)?   errorType: 'Meteor.Error'
I20200625-13:03:30.302(-4)? }
```

Reply:
```
I20200625-13:03:30.256(-4)? Setting Salesforce Bot Status, Response:
I20200625-13:03:30.257(-4)? {
I20200625-13:03:30.257(-4)?   statusCode: 200,
I20200625-13:03:30.257(-4)?   content: '{\n  "success": true\n}',
I20200625-13:03:30.257(-4)?   headers: {
I20200625-13:03:30.257(-4)?     'x-xss-protection': '1',
I20200625-13:03:30.257(-4)?     'x-instance-id': 'wrkqZy9JqnubebaXg',
I20200625-13:03:30.257(-4)?     'cache-control': 'no-store',
I20200625-13:03:30.257(-4)?     pragma: 'no-cache',
I20200625-13:03:30.257(-4)?     'content-type': 'application/json',
I20200625-13:03:30.257(-4)?     vary: 'Accept-Encoding',
I20200625-13:03:30.258(-4)?     date: 'Thu, 25 Jun 2020 17:03:30 GMT',
I20200625-13:03:30.258(-4)?     connection: 'close',
I20200625-13:03:30.258(-4)?     'transfer-encoding': 'chunked'
I20200625-13:03:30.258(-4)?   },
I20200625-13:03:30.258(-4)?   data: { success: true }
I20200625-13:03:30.290(-4)? }
```

As I can see here, it did set the user status as 'online' before the handoff. Maybe there is any mistake in the App configurations, can you please recheck them once?